pep-0706.rst

```diff
@@ -0,0 +1,509 @@
+PEP: 706
+Title: Filter for tarfile.extractall
+Author: Petr Viktorin <encukou@gmail.com>
+Discussions-To: 
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 09-Feb-2023
+Python-Version: 3.12
+Post-History: `25-Jan-2023 <https://discuss.python.org/t/23149>`__,
+
+
+Abstract
+========
+
+The extraction methods in ``tarfile`` gain a ``filter`` argument,
+which allows rejecting files or modifying metadata as the archive is extracted.
+Three built-in named filters are provided, aimed at limiting features that
+might be surprising or dangerous.
+These can be used as-is, or serve as a base for custom filters.
+
+After a deprecation period, a strict (but safer) filter will become the default.
+
+The API is meant to set a precedent for other archive libraries, such as
+``zipfile``.
+
+
+Motivation
+==========
+
+The ``tar`` format is used for several use cases, many of which have different
+needs. For example:
+
+- A backup of a UNIX workstation should faithfully preserve all kinds of
+  details like file permissions, symlinks to system configuration, and various
+  kinds of special files.
+- When unpacking a data bundle, it’s much more important that the unpacking
+  will not have unintended consequences – like exposing password file by
+  symlinking it to a public place.
+
+To support all its use cases, the ``tar`` format has many features.
+In many cases, it's best to ignore or disallow some of them when extracting
+an archive.
+
+Python allows extracting ``tar`` archives using
+``tarfile.TarFile.extractall()``, whose docs warn to
```

Comment:
Thanks for the suggestion, but it should be a link to the 3.11 version of the docs. If I recall correctly, the syntax for that is pretty ugly.

Reply:
> . If I recall correctly, the syntax for that is pretty ugly.

Not really—you just prefix the name of the Intersphinx target. Assuming we added Intersphinx targets with names `pyX.Y` for the relevant versions, it would just be

```suggestion
:meth:`py3.11:tarfile.TarFile.extractall()`, whose docs warn to
```

Even without that, as a reader it still seems much more valuable to implement the cross reference than not do so, as otherwise any interested readers will just have to manually google it and either get the latest version anyway (or whatever version their search engine happens to return), or worse some crap W3Schools or GeeksForGeeks article farm page.

Reply:
Hm. According to the [intersphinx documentation](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html) it seems it should be `` :external+py3.11:meth:`tarfile.TarFile.extractall()` ``. Is the form with the `py3.11` in the backticks documented elsewhere?

I'd hope PEP readers can avoid Google's suggestions. But OK, let me add the intersphinx mapping and linkify everything.

Reply:
I've added the intersphinx links. Maybe a bit of overkill, but without linking directly to 3.11, the links wouldn't be much better than letting people search for the terms themselves.

Reply:
> Hm. According to the [intersphinx documentation](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html) it seems it should be `` :external+py3.11:meth:`tarfile.TarFile.extractall()` ``. Is the form with the `py3.11` in the backticks documented elsewhere?

Hmm, yeah, I noticed that in double-checking for your previous reply above. The intersphinx extension reference docs only mentioned the `external` role added by the extension (which explicitly requests lookup via Intersphinx rather than in the docs' own index), and doesn't really go into the support for naming specific intersphinx targets in the internal roles—and I'm not really sure where that _is_ discussed, as I couldn't find it—it's possible I just learned it from seeing it in other docs sites (whereas I've never seen `external` actually used anywhere).

> Maybe a bit of overkill, but without linking directly to 3.11, the links wouldn't be much better than letting people search for the terms themselves.

I definitely agree it's better to link to a specific version for posterity, in line with PEPs being historical documents and all. Thanks for adding the mapping!